utils/ramses-scene-viewer/src/SceneViewer.cpp

```diff
@@ -125,9 +127,16 @@ namespace ramses_internal
         displayManager.setSceneMapping(loadedScene->getSceneId(), displayId);
         displayManager.setSceneState(loadedScene->getSceneId(), ramses_display_manager::SceneState::Rendered);
 
+        if(m_takeScreenshotArgument)
+        {
+            m_screenshotSaver = std::make_unique<ScreenshotSaver>(ScreenshotSaver(renderer, displayId, 800, 600));
+            m_rendererEventHandler.setScreenshotSaver(m_screenshotSaver.get());
+            m_rendererEventHandler.setTakeScreenshot(true);
+        }
+
         while (displayManager.isRunning())
         {
-            displayManager.dispatchAndFlush();
```

Comment:
better to put the "if screenshot enabled" check here and only pass in the m_rendererEventHandler if the option is enabled